src/pages/organize/[orgId]/projects/[campId]/emails/[emailId]/insights.tsx

```diff
@@ -0,0 +1,153 @@
+import { GetServerSideProps } from 'next';
+import Head from 'next/head';
+import { Paper, Typography } from '@mui/material';
+import { Box, useTheme } from '@mui/system';
+import { ResponsiveLine } from '@nivo/line';
+import { linearGradientDef } from '@nivo/core';
+import { FormattedTime } from 'react-intl';
+
+import EmailLayout from 'features/emails/layout/EmailLayout';
+import { PageWithLayout } from 'utils/types';
+import { scaffold } from 'utils/next';
+import useEmail from 'features/emails/hooks/useEmail';
+import useEmailStats from 'features/emails/hooks/useEmailStats';
+import { useNumericRouteParams } from 'core/hooks';
+import useServerSide from 'core/useServerSide';
+import ZUICard from 'zui/ZUICard';
+import ZUINumberChip from 'zui/ZUINumberChip';
+import EmailKPIChart from 'features/emails/components/EmailKPIChart';
+import useEmailInsights from 'features/emails/hooks/useEmailInsights';
+import ZUIFuture from 'zui/ZUIFuture';
+import { useMessages } from 'core/i18n';
+import messageIds from 'features/emails/l10n/messageIds';
+
+export const getServerSideProps: GetServerSideProps = scaffold(
+  async () => {
+    return {
+      props: {},
+    };
+  },
+  {
+    authLevelRequired: 2,
+    localeScope: ['layout.organize.email', 'pages.organizeEmail'],
+  }
+);
+
+const EmailPage: PageWithLayout = () => {
+  const messages = useMessages(messageIds);
+  const { orgId, emailId } = useNumericRouteParams();
+  const theme = useTheme();
+  const { data: email } = useEmail(orgId, emailId);
+  const stats = useEmailStats(orgId, emailId);
+  const insightsFuture = useEmailInsights(orgId, emailId);
+
+  const onServer = useServerSide();
+
+  if (onServer || !email) {
+    return null;
+  }
+
+  return (
+    <>
+      <Head>
+        <title>{email.title}</title>
+      </Head>
+      <ZUICard
+        header={messages.insights.opened.header()}
+        status={
+          <ZUINumberChip
+            color={theme.palette.grey[200]}
+            value={stats.numOpened}
+          />
+        }
+      >
+        <Box display="flex" gap={1}>
+          <Box flexGrow={0} maxWidth={300}>
+            <EmailKPIChart
+              email={email}
+              title={messages.insights.opened.gauge.header()}
+              total={stats.numSent}
+              value={stats.numOpened}
+            />
+            <Typography mb={2} mt={1} variant="body2">
+              {messages.insights.opened.description()}
+            </Typography>
+          </Box>
+          <Box flexGrow={1} height={550}>
+            <ZUIFuture future={insightsFuture}>
+              {(insights) => (
+                <ResponsiveLine
+                  animate={false}
+                  axisBottom={{
+                    format: '%b %d',
+                  }}
+                  colors={[theme.palette.primary.main]}
+                  curve="basis"
```

Comment:
I don't fully understand your reasoning here, maybe because I'm not sure whether you've looked at the output of [nivo line charts](https://nivo.rocks/line/) with the various options (your reference to the underlying d3 library makes me think that maybe you haven't).

If I understand correctly, you are arguing that we should not use `basis` because the line will not perfectly hit the control points. I think that is a fair argument, if the goal is to perfectly illustrate the change from point to point. But trying to do so in these extremely intense diagrams will most likely produce very jagged lines, which (IMO) distracts from the more important attribute of the chart which is to visualize the trend.

Reply:
You are completely right, both that we don't have local maximums/minimums, and that jaggedness is less of a problem for an ever-increasing line. But I still find that `monotoneX` is unnecessarily "jagged". I'm attaching two screenshots to show what I mean. So I think `basis` is actually fine (and better) for this.

## With `monotoneX`
![image](https://github.com/user-attachments/assets/a8f16900-91b2-468b-9d9f-e66639ecd92b)

## With `basis`
![image](https://github.com/user-attachments/assets/9c0ad886-099c-453a-be25-eac98ba48469)

Reply:
If the data is jagged, I think it's better to present at jagged line, unless you are trying to model some other underlying metric which is reasonably smooth.

Can you produce a third comparison screenshot with just the data points, without interpolation?

Reply:
> If the data is jagged, I think it's better to present at jagged line, unless you are trying to model some other underlying metric which is reasonably smooth.

I am trying to indicate a trend, not visualize exact values at exact times. It could be argued then that the trend could be made visible by sampling the data at even intervals (e.g. one point per hour), but instead I have taken the easier route of presenting all data but with a smoothed line. I hope it's clear from the screenshots above that the choice between `monotoneX` and `basis` is, in the end, mostly one of personal taste.

> Can you produce a third comparison screenshot with just the data points, without interpolation?

Grabbing a new screenshot is actually a fair bit of work. I feel like this a matter of such detail that I'd rather not spend the time doing that.